haystack/components/converters/tika.py

```diff
@@ -85,7 +117,11 @@ def run(
                 logger.warning("Could not read {source}. Skipping it. Error: {error}", source=source, error=e)
                 continue
             try:
-                text = tika_parser.from_buffer(io.BytesIO(bytestream.data), serverEndpoint=self.tika_url)["content"]
+                parsed = tika_parser.from_buffer(
+                    io.BytesIO(bytestream.data), serverEndpoint=self.tika_url, xmlContent=True
```

Comment:
Why this change to output xmlContent @lambda-science ? I'm afraid how it will affect the existing clients...

Would you please elaborate on your line fo thinking here?

Reply:
> Why this change to output xmlContent @lambda-science ? I'm afraid how it will affect the existing clients...
> 
> Would you please elaborate on your line fo thinking here?

Part of the answer is here: https://github.com/deepset-ai/haystack/issues/7949#issuecomment-2247983303

The PR code is an almost exact replicate of the code of TikaConveter in Haystack 1.X.
Without xmlContent, the text is parsed without page indication. Page break are either `\n\n` or `\n\n\n` but this can also happens in paragraph, so it's not enough to find the page break.

With xmlContent, page are delimited by HTML div with "page" class (or something like that) `<div class=page>`. This way we can parse the xmlContent to find the page HTML tags, and replace them by `\f`.
This shouldn't change the existing clients as what we do here is getting the results in XML format, removing all XML tags (with the Parser class on top) and replace specific XML tags about new pages by `\f`

Reply:
Riiight, thank you 🙏